### PR TITLE
create the demo task and server

### PR DESCRIPTION
**Pull request by nongfushanquan**
1.Run demoserver in format like ./demoserver ./data 100000, some files will generate under the folder ./data, 100000 is the recorder number in all files. the file number is in random.Then the server  which is  waiting for the rpc call form the client will listen the port 127.0.0.1:1234. 
2.Run demotask  in format like ./demotask  ./data ./data1.  the ./data is the upstream source, and the ./data2 is the destination folder. the demotask will send request to the demoserver , list the ./data first, get all the files under the upstream folder .As an example ,only sync the first file to the destination folder.
3. the demoserver will be updated to provide some args as to the ip later.
4. the demotask is just to show the workflow of invoking the sync process. we can implement it in the jobmaster\task mode later.

### CODE REVIEW

**Comment by hanfei1991** [comment]
Could you add some comments to explain how the demo works?

File: cmd/demotask/demotask.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			log.Fatal(err)
+		}
+		fmt.Printf("read the string %v \n", linestr.Linestr)
+		strs := strings.Split(linestr.Linestr, ",")
+		if len(strs) >= 2 {
+			c.buffer <- kv{key: strs[0], value: strs[1]}
+		} else {
+			break
+		}
+		time.Sleep(time.Second)
```
Comment (amyangfei) [inline cmd/demotask/demotask.go:81]: why sleep 1s here

File: cmd/demoserver/kvdatamake.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+		fileNum = r.Intn(FILENUM)
+	}
+	fmt.Printf("the fold file number is  %d  ", fileNum)
+	fileWriterMap := make(map[int]*bufio.Writer)
+	for i := 0; i < fileNum; i++ {
+		fileName := folderName + "/" + strconv.Itoa(i) + ".txt"
+		file, err := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY, 0o666)
+		if err != nil {
+			fmt.Printf("make sure the file % s exist, the error is %v ", fileName, err)
+		}
```
[the remark this answers is not in the document]
Comment (nongfushanquan) [reply cmd/demoserver/kvdatamake.go]: fix it

File: cmd/demoserver/kvdatamake.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		res, err := stream.Recv()
+		if err == nil {
+			fileName := res.FileName
+			//	fmt.Println("the string is ", fileName, res.Key, res.Value)
+			writer, exist := s.fileWriterMap[fileName]
+			if !exist {
+				file, err1 := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY, 0o666)
+				if err1 != nil {
+					fmt.Printf("make sure the file % s exist, the error is %v ", fileName, err1)
+				}
```
[the remark this answers is not in the document]
Comment (nongfushanquan) [reply cmd/demoserver/kvdatamake.go]: OK , I will fix it

File: cmd/demotask/demotask.go (hunk 1 above)
Comment (nongfushanquan) [reply cmd/demotask/demotask.go]: just for debug, remove it already

**Review by amyangfei — APPROVED** [review]

**Pull request merged**